### PR TITLE
SOLR-15965 Use proper signatures for SolrAuth

### PR DESCRIPTION
**Pull request by madrob**
https://issues.apache.org/jira/browse/SOLR-15965

This is a draft of what the code _could_ look like in 9.0 if we want to maintain backwards compatibility along the full 9x line. Alternatively, the majority of this code could go to 8.11.2 and only the "V2" portions of this code would make it into 9.0

### CODE REVIEW

**Comment by madrob** [comment]
I don't know why precommit failed on this building the ref guide with broken links, the link appears to work for me locally when I build it... @ctargett any ideas? Alternatively, we can wait until Antora comes in and then I'll redo the content anyway.

File: solr/solr-ref-guide/src/major-changes-in-solr-9.adoc [hunk 1]
```diff
@@ -53,7 +53,11 @@ SolrCloud doesn't support that but may sometime.
...
 This includes API calls for the replication handler and metrics.
-For rolling upgrades into 9.0, you need to be on Solr version 8.7 or greater. Some metrics also changed, alerts and  monitors on Solr KPIs that mention "master" or "slave" will also now be "leader" and "follower"
+For rolling upgrades into 9.0, you need to be on Solr version 8.7 or greater. Some metrics also changed, alerts and monitors on Solr KPIs that mention "master" or "slave" will also now be "leader" and "follower"
+
+* SOLR-15965: Internode communication secured by PKI Authentication has changed formats. To maintain compatability, you will need to either:
+** Upgrade the cluster to 8.11.2 or newer before proceeding to 9.0
+** Set system properties `solr.pki.sendVersion=v1` and `solr.pki.acceptVersions=v1,v2` on the upgraded nodes. After the whole cluster has been upgraded, you can optionally remove them and perform an additional rolling restart to reject requests coming in via the old format. For more information, see <<authentication-and-authorization-plugins.adoc#_pkiauthenticationplugin,PKI Authentication Plugin>>.
```
Comment (ctargett) [inline solr/solr-ref-guide/src/major-changes-in-solr-9.adoc:60]: The link format here is what's causing your precommit to fail. Our settings remove the asciidoctor default of inserting `_` before anchors, so all you need is `#pkiauthenticationplugin` without the underscore.

Comment (madrob) [reply solr/solr-ref-guide/src/major-changes-in-solr-9.adoc:60]: 🤦

File: solr/solr-ref-guide/src/major-changes-in-solr-9.adoc [hunk 2]
```diff
@@ -53,7 +53,11 @@ SolrCloud doesn't support that but may sometime.
 
 * SOLR-14702: All references to "master" and "slave" were replaced in the code with "leader" and "follower".
 This includes API calls for the replication handler and metrics.
-For rolling upgrades into 9.0, you need to be on Solr version 8.7 or greater. Some metrics also changed, alerts and  monitors on Solr KPIs that mention "master" or "slave" will also now be "leader" and "follower"
+For rolling upgrades into 9.0, you need to be on Solr version 8.7 or greater. Some metrics also changed, alerts and monitors on Solr KPIs that mention "master" or "slave" will also now be "leader" and "follower"
+
+* SOLR-15965: Internode communication secured by PKI Authentication has changed formats. To maintain compatability, you will need to either:
+** Upgrade the cluster to 8.11.2 or newer before proceeding to 9.0
```
Comment (tflobbe) [inline solr/solr-ref-guide/src/major-changes-in-solr-9.adoc:59]: I'm guessing we'll want 8.11.2 to default to `solr.pki.sendVersion=v1` and `solr.pki.acceptVersions=v1,v2` to support rolling upgrades from versions <= 8.11.1, right? if that's the case, even upgrading from 8.11.2 may require setting `solr.pki.acceptVersions=v1,v2` unless the user has already upgraded to v2 explicitly.

**Review by tflobbe — APPROVED** [review]
LGTM!

File: solr/solr-ref-guide/src/major-changes-in-solr-9.adoc (hunk 2 above)
Comment (madrob) [reply solr/solr-ref-guide/src/major-changes-in-solr-9.adoc]: I tried making the upgrade instructions more explicit. Let me know what you think.

**Pull request merged**